Tasks/MavenAuthenticateV0/mavenutils.ts

```diff
@@ -0,0 +1,200 @@
+                    configuration: {
+                        httpHeaders: {
+                            property: {
+                                name: 'Authorization',
+                                value: 'Basic ' + tokenAuthInfo.token
+                            }
+                        }
+                    }
+                });
+                tl.debug(`Detected token credentials for '${serviceConnection.packageSource.uri}'`);
+                break;
+            case (ServiceConnectionAuthType.PrivateKey):
+                const privateKeyAuthInfo = serviceConnection as PrivateKeyServiceConnection;
+                serverElements.push({
+                    id: serviceConnection.additionalData["REPOSITORYID"],
+                    privateKey: privateKeyAuthInfo.privateKey,
+                    passphrase: privateKeyAuthInfo.passphrase
+                });
+                tl.debug(`Detected token credentials for '${serviceConnection.packageSource.uri}'`);
+                break;
+            default:
+                throw Error(tl.loc('Error_InvalidServiceConnection', serviceConnection.packageSource.uri));
+        }
+    }   
+
+    return serverElements;
+}
+
+export function readXmlFileAsJson(filePath: string): Q.Promise<any> {
+    return readFile(filePath, 'utf-8')
+        .then(convertXmlStringToJson);
+}
+
+export function writeJsonAsSettingsFile(filePath: string, jsonContent: any): Q.Promise<void> {
+    return writeJsonAsXmlFile(filePath, jsonContent.settings, 'settings');
+}
+
+export function mavenSettingsJsonInsertServer (json: any, serverJson:any): any {
+    if (!json) {
+        json = {};
+    }
+    if (!json.settings || typeof json.settings === "string") {
+        json.settings = {};
+    }
+    if (!json.settings.$) {
+        json.settings.$ = {};
+        json.settings.$['xmlns'] = 'http://maven.apache.org/SETTINGS/1.0.0';
+        json.settings.$['xmlns:xsi'] = 'http://www.w3.org/2001/XMLSchema-instance';
+        json.settings.$['xsi:schemaLocation'] = 'http://maven.apache.org/SETTINGS/1.0.0' + os.EOL + 'https://maven.apache.org/xsd/settings-1.0.0.xsd';
+    }
+    if (!json.settings.servers) {
+        json.settings.servers = {};
+    }
+    addPropToJson(json.settings.servers, 'server', serverJson);
+    return json;
+}
+
+
+function addPropToJson(obj: any, propName:string, value: any): void {
+    if (!obj) {
+        obj = {};
+    }
+
+    if (obj instanceof Array) {
+        let propNode = obj.find(o => o[propName]);
+        if (propNode) {
+            obj = propNode;
+        }
+    }
+
+    let containsId: (o) => boolean = function(o) {
+        if (value && value.id) {
+            if (o.id instanceof Array) {
+                return o.id.find((v) => {
+                    return v === value.id;
+                });
+            } else {
+                return value.id === o.id;
+            }
+        }
+        return false;
+    };
+
+    if (propName in obj) {
```

Comment:
nit: this block of code is a little hard to read for me. Consider simplifying it if you feel the same. May be just put some comments explaining the purpose , better variable names, reduce nesting of if/else.

Reply:
Fixed